cpp/test/cluster/kmeans.cu

```diff
@@ -0,0 +1,181 @@
+    params.tol                 = testparams.tol;
+    params.n_init              = 5;
+    params.rng_state.seed      = 1;
+    params.oversampling_factor = 0;
+
+    auto stream = handle.get_stream();
+    auto X      = raft::make_device_matrix<T>(n_samples, n_features, stream);
+    auto labels = raft::make_device_vector<int>(n_samples, stream);
+
+    raft::random::make_blobs<T, int>(X.data(),
+                                     labels.data(),
+                                     n_samples,
+                                     n_features,
+                                     params.n_clusters,
+                                     stream,
+                                     true,
+                                     nullptr,
+                                     nullptr,
+                                     T(1.0),
+                                     false,
+                                     (T)-10.0f,
+                                     (T)10.0f,
+                                     (uint64_t)1234);
+
+    d_labels.resize(n_samples, stream);
+    d_labels_ref.resize(n_samples, stream);
+    d_centroids.resize(params.n_clusters * n_features, stream);
+
+    std::optional<raft::device_vector_view<const T>> d_sw = std::nullopt;
+    auto d_centroids_view =
+      raft::make_device_matrix_view<T>(d_centroids.data(), params.n_clusters, n_features);
+    if (testparams.weighted) {
+      d_sample_weight.resize(n_samples, stream);
+      d_sw = std::make_optional(
+        raft::make_device_vector_view<const T>(d_sample_weight.data(), n_samples));
+      thrust::fill(thrust::cuda::par.on(stream),
+                   d_sample_weight.data(),
+                   d_sample_weight.data() + n_samples,
+                   1);
+    }
+
+    raft::copy(d_labels_ref.data(), labels.data(), n_samples, stream);
+    handle.sync_stream(stream);
+
+    T inertia   = 0;
+    int n_iter  = 0;
+    auto X_view = (raft::device_matrix_view<const T>)X.view();
+
+    raft::cluster::kmeans_fit_predict<T, int>(
+      handle,
+      params,
+      X_view,
+      d_sw,
+      d_centroids_view,
+      raft::make_device_vector_view<int>(d_labels.data(), n_samples),
+      raft::make_host_scalar_view(&inertia),
+      raft::make_host_scalar_view(&n_iter));
+
+    handle.sync_stream(stream);
+
+    score = raft::stats::adjusted_rand_index(
+      d_labels_ref.data(), d_labels.data(), n_samples, handle.get_stream());
+
+    if (score < 1.0) {
+      std::stringstream ss;
+      ss << "Expected: " << raft::arr2Str(d_labels_ref.data(), 25, "d_labels_ref", stream);
+      std::cout << (ss.str().c_str()) << '\n';
+      ss.str(std::string());
+      ss << "Actual: " << raft::arr2Str(d_labels.data(), 25, "d_labels", stream);
+      std::cout << (ss.str().c_str()) << '\n';
+      std::cout << "Score = " << score << '\n';
+    }
+  }
+
+  void SetUp() override { basicTest(); }
+
+ protected:
+  cudaStream_t stream = 0;
```

Comment:
This test predates the use of the handle in the testing layer. We should create a handle here and use the stream from that instead of creating an explicit stream. Also less prone to errors.